sbi/inference/posteriors/base_posterior.py

```diff
@@ -129,6 +129,7 @@ def set_default_x(self, x: Tensor) -> "NeuralPosterior":
         self._x = process_x(
             x, x_shape=self._x_shape, allow_iid_x=self.potential_fn.allow_iid_x
         ).to(self._device)
+        self._map = None
```

Comment:
This used to be called `._map`. If we rename it, we might have to ensure backwards compatibility. @janfb what do you think? Is it ok to just rename it or should we require backwards compatibility?

Or should we just stick to the name `self.map_`?